ts/components/ScreenHeader.tsx

```diff
@@ -78,7 +79,7 @@ class ScreenHeader extends React.Component<Props> {
         <View accessible={true} style={styles.text}>
           {heading}
         </View>
-        {this.getIcon()}
+        {fromNullable(rightComponent).fold(this.getIcon(), c => c)}
```

Comment:
I actually understand your concern, and I agree with you that we should be more explicit, I'll add a comment to specify that the custom rightComponent will be rendered even if an icon name is specified.

I preferred to avoid a complete refactoring of the component with a single prop specifying the right item to render cause of the wide usage of ScreenContentHeader/ScreenHeader. What do you think?

Reply:
Fair enough :-) I'm fine with adding the comment in the component's doc.